modules/dnn/src/cuda4dnn/csl/kernel_utils.hpp

```diff
@@ -0,0 +1,153 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_DNN_CUDA4DNN_CSL_KERNEL_UTILS_HPP
+#define OPENCV_DNN_CUDA4DNN_CSL_KERNEL_UTILS_HPP
+
+#include "error.hpp"
+#include "stream.hpp"
+#include "nvcc_defs.hpp"
+
+#include <cstddef>
+
+#ifdef __CUDACC__
+#include <cuda_runtime.h>
+#endif
+
+namespace cv { namespace dnn { namespace cuda4dnn { namespace csl {
+
+#ifdef __CUDACC__
+    struct execution_policy {
```

Comment:
Contrary to my thoughts while I gave this name, I now feel that this name does more harm than good.  When OpenCV switches to C++17, this `execution_policy` will cause confusion with `std::execution_policy`. Even though they are sort of related, I think it's better off to distinguish the two clearly.

Reply:
Yeah, using a different name is not a bad idea.